server/api/crud/events.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import datetime
+
+import sqlalchemy.orm
+
+import mlrun.utils.singleton
+import server.api.api.utils
+import server.api.utils.singletons.db
+from mlrun.utils import logger
+
+
+class Events(
+    metaclass=mlrun.utils.singleton.Singleton,
+):
+    _cache = {}
+
+    @staticmethod
+    def is_valid_event(project: str, event_data: mlrun.common.schemas.Event):
+        if event_data.entity.project != project:
+            return False
+
+        return bool(event_data.is_valid())
+
+    def add_event(self, project, name, alert_name):
+        self._cache.setdefault((project, name), []).append(alert_name)
+
+    def remove_event(self, project, name):
+        del self._cache[(project, name)]
+
+    def delete_alert_events(self, project):
+        to_delete = [name for proj, name in self._cache if proj == project]
```

Comment:
```suggestion
        to_delete = [name for proj, name in self._cache.items() if proj == project]
```

`in self._cache` would return only the keys and this line would raise an exception. Do you have a unit test covering this API?

Reply:
actually, there is a test for exactly this case. the test is called test_alert_after_project_deletion() in integration tests
the reason this works is that the key is actually (project, name)